website/docs/r/stack_hci_virtual_hard_disk.html.markdown

```diff
@@ -0,0 +1,99 @@
+---
+subcategory: "Azure Stack HCI"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_stack_hci_virtual_hard_disk"
+description: |-
+  Manages an Azure Stack HCI Virtual Hard Disk.
+---
+
+# azurerm_stack_hci_virtual_hard_disk
+
+Manages an Azure Stack HCI Virtual Hard Disk.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-rg"
+  location = "West Europe"
+}
+
+resource "azurerm_stack_hci_storage_path" "example" {
+  name                = "example-sp"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  custom_location_id  = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ExtendedLocation/customLocations/cl1"
+  path                = "C:\\ClusterStorage\\UserStorage_2\\sp-example"
+  tags = {
+    foo = "bar"
+  }
+}
+
+resource "azurerm_stack_hci_virtual_hard_disk" "example" {
+  name                = "example"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  custom_location_id  = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ExtendedLocation/customLocations/cl1"
```

Comment:
Same here.. Assuming this can be different that the storage path's location id?

Reply:
The `custom_location_id` must be same to the storage path's.

Reply:
Then, is it possible to only provide the `storage_path_id`, then read it out from the storage path?

Reply:
Do you mean to read the storage path and get the `custom_location_id` before creating the virtual disk? Maybe we should not do this because it needs extra permissions to read storage path.

Reply:
Can you elaborate on the "extra permissions"? The point is to avoid the case that users specify different `custom_location_id` for the storage path and the virtual disk. Also it improves the UX.

Reply:
"extra permissions" is the specific permission to Get/List storage path. This permission is not needed if we only want to create the virtual disk. Although in terraform config environment, the client usually has these permissions..

Another reason is that `storage_path_id` is optional, if no `storage_path_id` is set, service backend will place the virtual disk on an available storage path. So in this case we cannot retrieve `custom_location_id` from the storage path.